### PR TITLE
Fixes #1167

### LINKED ISSUE #1167: REPL evaluation doesn't work after interrupting prover with Ctrl-C
This seems very closely related to #1157. Since #1158, cryptol can continue gracefully after a Ctrl-C during evaluation or while waiting for input. But the REPL still gets into a bad state if you interrupt it during a long-running `:prove` command. This example reliably causes a "resource vanished" error for me on MacOS:
```
Cryptol> :prove \(x:Z 257) -> x^^257 == x
^C
SBV exception:

*** Data.SBV: Interrupt:
***
***    Sent      : (check-sat)
***
***    Executable: /Users/huffman/.bin/z3
***    Options   : -nw -in -smt2

Cryptol> True
cryptol: fd:13: hFlush: resource vanished (Broken pipe)
```

### PR DESCRIPTION
**Pull request by yav**
This requires newer version simple-smt, which supports calling back into
the program when the solver exits

### CODE REVIEW

File: src/Cryptol/REPL/Monad.hs [hunk 1]
```diff
@@ -415,13 +415,18 @@ getPrompt  = mkPrompt `fmap` getRW
 getCallStacks :: REPL Bool
 getCallStacks = eCallStacks <$> getRW
 
+clearTCSolverAction :: REPL (IO ())
+clearTCSolverAction =
+  REPL (\ref -> pure (modifyIORef ref (\rw -> rw { eTCSolver = Nothing })))
```
Comment (robdockins) [inline src/Cryptol/REPL/Monad.hs:420]: Perhaps we should use `atomicModifyIORef` here, since this may occur asynchronously?

**Review by robdockins — APPROVED** [review]
Seems like a reasonable solution to me, although I haven't tested that it works.

We should probably remove the other handler, though.

**Comment by robdockins** [comment]
@yav, any objections if I rebase this and get it merged?

**Comment by yav** [comment]
@robdockins nope, go for it, I was waiting for the CI to get sorted out, but I don't think the failures are important.

**Pull request merged**